gcp_variant_transforms/transforms/merge_header_definitions.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2018 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Beam combiner function for merging VCF file header definitions."""
+
+from collections import OrderedDict  # pylint: disable=unused-import
+from collections import namedtuple
+from typing import Dict, Union  # pylint: disable=unused-import
+
+import apache_beam as beam
+from gcp_variant_transforms.beam_io.vcf_header_io import VcfHeader   # pylint: disable=unused-import
+from gcp_variant_transforms.libs.vcf_field_conflict_resolver import VcfParserConstants
+
+HeaderFields = namedtuple('HeaderFields', [VcfParserConstants.NUM,
+                                           VcfParserConstants.TYPE])
+
+
+class DefinitionDetails(object):
+  """Container for conflict details for one field ID."""
+
+  # For the same field definition, save at most `_MAX_NUM_FILE_NAMES` names.
+  _MAX_NUM_FILE_NAMES = 5
+
+  def __init__(self, definition, file_name):
+    # type: (OrderedDict[str, Union[str, int]], str) -> None
+    """Initializes a _DefinitionDetails object.
+
+    Creates a dictionary that maps field definition to a list of file names.
+    Upon initialization, add ``definition`` and ``file_name`` to the dictionary.
+    """
+    self._definitions_to_files = {
+        HeaderFields(definition[VcfParserConstants.NUM],
+                     definition[VcfParserConstants.TYPE]): [file_name]}
+
+  def __eq__(self, other):
+    return self._definitions_to_files == other.definitions_to_files
+
+  def _add_details(self, other):
```

Comment:
As discussed offline: please move this inside the merge function as this is really implementing the merging logic. Should also rename the method as `add_details` is very generic (i dont have a good name for it now though :p )

Reply:
Done. Renamed `add_details` to `merge_definitions_to_files_dict`